modules/example-browser/src/caching_materials_manager.ts

```diff
@@ -0,0 +1,200 @@
+  KMS,
+  getClient,
+  encrypt,
+  decrypt,
+  WebCryptoCachingMaterialsManager,
+  getLocalCryptographicMaterialsCache
+} from '@aws-crypto/client-browser'
+import { toBase64 } from '@aws-sdk/util-base64-browser'
+
+/* This is injected by webpack.
+ * The webpack.DefinePlugin will replace the values when bundling.
+ * The credential values are pulled from @aws-sdk/credential-provider-node
+ * Use any method you like to get credentials into the browser.
+ * See kms.webpack.config
+ */
+declare const credentials: {accessKeyId: string, secretAccessKey:string, sessionToken:string }
+
+/* This is done to facilitate testing. */
+export async function testCachingMaterialsManagerExample () {
+  /* A KMS CMK is required to generate the data key.
+   * You need kms:GenerateDataKey permission on the CMK in generatorKeyId.
+   */
+  const generatorKeyId = 'arn:aws:kms:us-west-2:658956600833:alias/EncryptDecrypt'
+
+  /* Adding alternate KMS keys that can decrypt.
+   * Access to kms:Encrypt is required for every CMK in keyIds.
+   * You might list several keys in different AWS Regions.
+   * This allows you to decrypt the data in any of the represented Regions.
+   * In this example, I am using the same CMK.
+   * This is *only* to demonstrate how the CMK ARNs are configured.
+   */
+  const keyIds = ['arn:aws:kms:us-west-2:658956600833:key/b3537ef1-d8dc-4780-9f5a-55776cbb2f7f']
+
+  /* Need a client provider that will inject correct credentials.
+   * The credentials here are injected by webpack from your environment bundle is created
+   * The credential values are pulled using @aws-sdk/credential-provider-node.
+   * See kms.webpack.config
+   * You should inject your credential into the browser in a secure manner,
+   * that works with your application.
+   */
+  const { accessKeyId, secretAccessKey, sessionToken } = credentials
+
+  /* getClient takes a KMS client constructor
+   * and optional configuration values.
+   * The credentials can be injected here,
+   * because browser do not have a standard credential discover process the way Node.js does.
+   */
+  const clientProvider = getClient(KMS, {
+    credentials: {
+      accessKeyId,
+      secretAccessKey,
+      sessionToken
+    }
+  })
+
+  /* The KMS keyring must be configured with the desired CMKs */
+  const keyring = new KmsKeyringBrowser({ clientProvider, generatorKeyId, keyIds })
+
+  /* Create a cache to hold the material.
+   * In this case we use the local cache provided by the Encryption SDK.
+   * The number is the maximum number of entries that will be cached.
+   * Both encrypt and decrypt requests count independently towards this total.
+   * Elements will be actively removed from the cache.
+   * The default frequency is to check one item every minute.
+   * This can be configure by passing a `proactiveFrequency`
+   * as the second paramter to however often you want to check in milliseconds.
+   */
+  const cache = getLocalCryptographicMaterialsCache(100)
```

Comment:
A capacity value of 100 is very high. The typical value is 10. Also, I would make this a variable so it's meaning is clear. 

    const maxSize = 10
    const cache = getLocalCryptographicMaterialsCache(maxSize)

Reply:
will update in both places but I will use capacity